python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -672,17 +673,25 @@ def markInputUsed(self, quantum, ref):
                 raise KeyError("{} is not a predicted consumer for {}".format(ref, quantum))
             quantum._markInputUsed(ref)
 
-    def addDataUnit(self, unit, replace=False):
-        """Add a new `DataUnit`, optionally replacing an existing one
-        (for updates).
+    def addDataUnitEntry(self, dataUnitName, value):
+        """Add a new `DataUnit` entry.
 
-        unit : `DataUnit`
-            The `DataUnit` to add or replace.
-        replace : `bool`
-            If `True`, replace any matching `DataUnit` that already exists
-            (updating its non-unique fields) instead of raising an exception.
+        dataUnitName : `str`
+            Name of the `DataUnit` (e.g. ``"Camera"``).
+        values : `dict`
+            Dictionary of ``columnName, value`` pairs.
```

Comment:
Should document that `value` is itself a dictionary.

Reply:
It is? I had assumed it looked something like: `{"camera": "HSC", "abstract_filter": "i", "physical_filter": "HSC_i", "visit": 3}`. Did you have something else in mind?

Reply:
Oh, I was confused by the fact that the argument is called `values` in the doc but `value` in the code (and I was looking at the line where you do `**value` below).  Just rename the argument and I think it'll all make sense.